Automattic/vip-cli
# Drop Node 8 Support, Update Babel, & Support modern syntax across node versions

## Description

The main goal of this PR is to add language support for [optional chaining](https://github.com/tc39/proposal-optional-chaining/tree/8461024a040d9c41ec039e2d4291fe2dfda36574) & [nullish coalescing](https://github.com/tc39/proposal-nullish-coalescing/tree/5b2a58ee3a56c2b5a5eb30a30959e32ae61d4c49). Both are [Stage 4 proposals](https://tc39.es/process-document/) which means they "will be included in the soonest practical standard revision."

* Install `@babel/preset-env` dev dependency
* Use `@babel/preset-env` targeting node v8.9.0 (current >= target in package.json)
* Allow optional chaining & nullish coalescing in the flowconfig. 
* Target transpilation to node 10

Additionally, this drops support for node 8 as it's EOL'd.

## Steps to Test

1. Check out PR.
1. rm your `node_modules` folder
1. `npm i`
1. Run `npm run build`
    * There should be no build errors
1. Run some commands
    * They should behave identically as the upstream branch
1. Automated tests should pass
1. Built bundle size should not be drastically increased
